### DISCUSSION TITLE
Consideration: Change `nl2br()` to CSS: `white-space: pre-line`

### DISCUSSION THREAD

**Discussion by drbyte** [zencart/zencart#6139] [Ideas]
In the larger majority of places where we use `nl2br()` it's to pass the admin-entered (or language-pack-supplied) text content with linebreaks forward to the browser so that the browser also shows the line-breaks.

But, there's a CSS directive (`white-space: pre-line`) that will do that for us, meaning we could dump the majority of `nl2br()` wrappers, especially in templates (and template-output functions).

We could invoke it on the specific elements that output said data (like add the directive in multiple CSS declarations), or add one "utility class" helper and add that to the template elements that should use it, such as:
```css
.whitespace-pre-line {
    white-space: pre-line;
}
```

Seems to be supported on all major browsers since 2014: https://caniuse.com/?search=white-space%3A%20pre-line%3B

**Comment by lat9** [comment]
Browsers, likely per that posting, but email apps?

**Reply by drbyte** [reply]
Thanks. I forgot to mention that part.
No, definitely not reliable for email. Email clients are still stuck in the 1900's :(
